tests/test_check_unused_python_file.py

```diff
@@ -0,0 +1,60 @@
+import os
+from pathlib import Path
+from typing import List
+
+from oca_pre_commit_hooks.check_unused_python_file import main
+
+FILE_NAMES = ["res_partner", "project_task", "helpdesk_ticket"]
+
+
+def _gen_filepaths(filenames: List[str], basedir) -> List[str]:
+    return [f"{os.path.join(basedir, filename)}.py" for filename in filenames]
+
+
+def test_all_used_files(tmpdir):
```

Comment:
Why are you not using `unittest`?

Could you use it?

Reply:
Use a `setUp` to create tempdir and use a `tearDown` to delete it

Similar to https://github.com/Vauxoo/pre-commit-vauxoo/blob/332a706097926863d0235630fc3731a27ad4f559/tests/test_pre_commit_vauxoo.py#L21

and https://github.com/Vauxoo/pre-commit-vauxoo/blob/332a706097926863d0235630fc3731a27ad4f559/tests/test_pre_commit_vauxoo.py#L40-L41

Reply:
BTW I didn't find where `tmpdir` variable is assigned

Reply:
Since we are using pytest I did not find it necessary to use the old unittest way. Using a `setUp` and `tearDown` is not necessary since pytest already does this for us, that is why I am using `tmpdir` which automatically creates a temporary directory, that option is deprecated tho so I need to change it to `tmp_path`.

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

Reply:
Also, I find it cleaner to separate test files per functionality being tested instead of placing all tests in the same file so I placed it separately in order to raise a point about it (although I guess this is just personal preference). 

I was also thinking that perhaps our main way of testing as of right now (running and comparing outputs against a test files) is more of an integration test rather than a unit test, and perhaps we could keep our current tests (but treat them as integration tests), and write more modular unit tests like the one in this PR for each block/check.

Just an idea I think we could consider for the reestructuring